infra/hooks/postprovision.ps1

```diff
@@ -0,0 +1,6 @@
+if ($env:OPENAI_HOST -eq "azure") {  
+    $AZURE_OPENAI_API_KEY = az cognitiveservices account keys list --name $env:AZURE_OPENAI_SERVICE --resource-group $env:AZURE_RESOURCE_GROUP --query key1 --output tsv 
+    azd env set AZURE_OPENAI_API_KEY $AZURE_OPENAI_API_KEY  
+}  
+
+azd env get-values | Out-File -FilePath .env  
```

Comment:
Instead of putting the .env in the root of the repo I want to do this:

1. Whenever you need the .env values from the azd env, you find the config.json file in .azure folder, parse out the defaultEnvironment value, and then build the path to the .env  .azure/{defaultEnvironment}/config.json

And then use that path.


Something like this:

```
using DotNetEnv;

public class AzureEnvManager
{
    private const string ConfigFilePath = ".azure/config.json";

    public string GetDefaultEnvironment()
    {
        if (!File.Exists(ConfigFilePath))
        {
            throw new FileNotFoundException($"The config file was not found: {ConfigFilePath}");
        }

        var jsonData = File.ReadAllText(ConfigFilePath);
        var config = JObject.Parse(jsonData);
        var defaultEnvironment = config["defaultEnvironment"]?.ToString();

        if (string.IsNullOrEmpty(defaultEnvironment))
        {
            throw new InvalidOperationException("defaultEnvironment is not set in the config file.");
        }

        return defaultEnvironment;
    }

    public string GetEnvFilePath()
    {
        var defaultEnvironment = GetDefaultEnvironment();
        var envFilePath = Path.Combine(".azure", defaultEnvironment, ".env");

        if (!File.Exists(envFilePath))
        {
            throw new FileNotFoundException($"The .env file was not found: {envFilePath}");
        }

        return envFilePath;
    }

    public void LoadEnvVariables()
    {
        var envFilePath = GetEnvFilePath();
        Env.Load(envFilePath);

        // Example usage: accessing a loaded environment variable
        var exampleVar = Environment.GetEnvironmentVariable("EXAMPLE_VAR");
        Console.WriteLine($"EXAMPLE_VAR: {exampleVar}");
    }
}


```

Reply:
And please put that in a lib project that is shared amongst the other csproj files

Reply:
Updated.